cdk/transactionAnalyzer/src/main/java/software/amazon/event/kafkaconnector/TransactionAnalyzer.java

```diff
@@ -0,0 +1,164 @@
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.kafka.common.config.TopicConfig;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.kstream.KStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * This class reads records from an input topic (events), filters events which have a total of over 180 and sends those
+ * to an output topic (notifications) using Kafka Streams
+ */
+public class TransactionAnalyzer {
+    private static final Logger log = LoggerFactory.getLogger(TransactionAnalyzer.class);
+    private static final Properties props = new Properties();
+    private static final Schema.Parser parser = new Schema.Parser();
+    private static final CountDownLatch latch = new CountDownLatch(1);
+
+    public static void main(String[] args) {
+
+        var sourceTopic = System.getenv().getOrDefault("TOPIC_NAME", "events");
+        var notificationsTopic = System.getenv().getOrDefault("NOTIFICATIONS_TOPIC_NAME", "notifications");
+        var replicationFactor = System.getenv().getOrDefault("REPLICATION_FACTOR", "3");
+        var partitionCount = System.getenv().getOrDefault("PARTITION_COUNT", "10");
+        var notificationSchemaFile = System.getenv().getOrDefault("NOTIFICATION_SCHEMA_FILE", "notification.avsc");
+
+        var notificationSchema = createSchema(notificationSchemaFile);
+        var properties = setConfiguration();
+
+        var topicCreator = new TransactionAnalyzerTopicCreator(properties);
+        topicCreator.createTopic(notificationsTopic, Integer.valueOf(partitionCount), Short.valueOf(replicationFactor));
+        topicCreator.createTopic(sourceTopic, Integer.valueOf(partitionCount), Short.valueOf(replicationFactor));
+        topicCreator.close();
+
+        StreamsBuilder builder = new StreamsBuilder();
+        final KStream<String, GenericRecord> events = builder.stream(sourceTopic);
+
+        events
+                .filter((key, value) -> Double.parseDouble(value.get("total").toString()) > 195.00)
+                .map((userId, value) -> {
+                    var notification = createNotificationRecord(userId, value, notificationSchema);
+                    //As this is synthetic test data we can log any value as it does not contain PII
+                    log.info("Detected suspicious transaction. Sending notification to topic. UserId: {} Value: {}", userId, value.get("total").toString());
+                    return KeyValue.pair(UUID.randomUUID().toString(), notification);
```

Comment:
questions:

- why a random UUID, i.e., not reuse user id?
- what's this key used for? topic key?

Reply:
userId is already part of notification, using UUID to avoid any potential hot partitions. Didn't want to set it to null, so users can see how keys are sent to EventBridge.